sdk/identity/identity/src/credentials/managedIdentityCredential/index.ts

```diff
@@ -198,6 +198,11 @@ export class ManagedIdentityCredential implements TokenCredential {
 
         logger.getToken.info(`MSAL Identity source: ${identitySource}`);
 
+        // ServiceFabric does not support specifying user-assigned managed identity by client ID or resource ID. The managed identity selected is based on the resource configuration.
+        if (identitySource === "ServiceFabric" && (this.clientId || this.resourceId)) {
+          throw new CredentialUnavailableError(serviceFabricErrorString);
```

Comment:
Shouldn't this be fatal for credential chains i.e., not this type? Specifying an identity implies a desire to authenticate that identity and an expectation that will work

Reply:
That's a good point. Per @maorleger comment above, I'll update the logic to catch this error early and throw directly in the constructor. From my understanding, that will be fatal to the credential chains right away.